hw/ip/spi_device/dv/env/spi_device_scoreboard.sv

```
@@ -206,6 +252,38 @@ class spi_device_scoreboard extends cip_base_scoreboard #(.CFG_T (spi_device_env
     end
   endfunction
 
+  // Check if opcode is enabled
+  virtual function check_opcode_enable(bit [7:0] q_opcode, ref bit enable, ref bit [4:0] en_idx);
```

This function returns the index of the highest entry that's enabled and silently ignores the lower entries if two or more are enabled.

Is that what you intend? If you know that there's only ever at most one that's enabled, maybe add a check? If you explicitly want this "highest wins" behaviour, I think there should probably be a comment at the top of the function that explains what it does.

I think a better API would probably be to return the bit mask of commands that are enabled (so lines 261 and 262 get replaced with something like `ret = ret | (32'd1 << i)`). That way, there are no confusing `ref` arguments. Callers can check whether something is enabled with a unary OR or comparing with zero.

---

This checks if specific opcode is configured in any of cmd_info registers and in addition checks if it is valid (enabled). I think it is irregular to have same opcodes configured to 2 different cmd_info registers  - @eunchan can you confirm. This can possibly lead to configuring same command with different behaviour? Which cmd_info behaviour wins if this is the case?
If there are more cmd_info registers configured with the same opcode your observation is correct, probably some checker has to be added so we have this irregularity reported.

---

Adding @eunchan to Reviewers list - please comment on this situation of configuring same opcode in two different cmd_info registers.

---

yep. Hardware does exactly like the function described here. And the expectation is there's no two command information slots with valid have same opcode.

---

Cool! In which case, could you add a note at the top of the function explaining what it does and that it matches the HW? (And I guess we need some functional coverage somewhere for the situation where there's more than one valid opcode, but that obviously doesn't need to be in this PR)